src/shared/test/injector-test.js

```diff
@@ -66,7 +66,7 @@ describe('Injector', () => {
 
       container.register('foo', fooFactory);
       container.register('bar', barFactory);
-      container.register('baz', bazFactory);
+      container.register('baz', { factory: bazFactory });
```

Comment:
Again, I really want to understand this better...

Reply:
Ah, I should clarify. `bazFactory` is declared with an arrow function above, but `register(...)` by default assumes its second argument is a class and tries to call `new` on it. The easiest way to see the problem is to try reverting this to the old code and running the test.